google-cloud-debugger/lib/google/cloud/debugger.rb

```diff
@@ -402,13 +402,61 @@ def self.new project_id: nil, credentials: nil, service_name: nil,
       # for full configuration parameters.
       #
       # @return [Stackdriver::Core::Configuration] The configuration object
-      #   the Google::Cloud::ErrorReporting module uses.
+      #   the Google::Cloud::Debugger module uses.
       #
       def self.configure
         yield Google::Cloud.configure[:debugger] if block_given?
 
         Google::Cloud.configure[:debugger]
       end
+
+      ##
+      # Allow calling of potentially state-changing methods even if mutation
+      # detection is configured to be active.
+      #
+      # Generally it is unwise to run code that may change the program state
+      # (e.g. modifying instance variables or causing other side effects) in a
+      # breakpoint expression, because it could change the behavior of your
+      # program. However, the checks are currently quite conservative, and may
+      # block code that is actually safe to run. If you are certain your
+      # expression is safe to evaluate, you may wrap it in a block using this
+      # method. Here is an example expression:
+      #
+      # ```ruby
```

Comment:
You should be able to use YARD's `@example` here, in order to support the Angular docs app.

Reply:
Do we have an examples tester that will be triggered? This method will not work in a normal environment; it works only when called from an expression evaluation, which is really difficult to simulate in a test harness.

Also, does it do the right thing if there are multiple `@example` blocks interleaved with text (which is what I have here)? It was a while ago that I worked with the docs app, but I thought it tried to combine multiple examples.

Reply:
> Do we have an examples tester that will be triggered?

Yes. Run `rake doctest` to see if the examples fail, and then skip them as follows in `support/doctest_helper.rb` if it is not possible to mock them:
 
```ruby
doctest.skip "Google::Cloud::Debugger::Project#attach"
```

Reply:
No, you can only have a single-line "title" on the same line as the `@example` tag. We only interleave text and examples in the top-level "guide" (located on the `Debugger` module in this file). For all other class and method docs, the complete discussion is followed by the examples. If there is more than one example, the titles refer back to concepts/usages in the discussion.

Reply:
Here's an example from Logging: https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-logging/lib/google/cloud/logging/middleware.rb#L127-L173

Reply:
Thanks @quartzmo for the help! Updated.